flocker/volume/service.py

```diff
@@ -10,27 +10,37 @@
 import sys
 import json
 import stat
+from subprocess import check_call
 from uuid import UUID, uuid4
+import time
+
+import ipaddr
 
 from zope.interface import Interface, implementer
 
 from characteristic import attributes
 
-from twisted.internet.defer import maybeDeferred
+from twisted.internet.defer import maybeDeferred, succeed
 from twisted.internet.task import deferLater
 from twisted.python.filepath import FilePath
 from twisted.application.service import Service
 from twisted.internet.defer import fail
 
+# The comment below, and the issue link in particular, doesn't make
+# much sense to me. Wrong issue link?
 # We might want to make these utilities shared, rather than in zfs
 # module... but in this case the usage is temporary and should go away as
 # part of https://clusterhq.atlassian.net/browse/FLOC-64
-from .filesystems.zfs import StoragePool
-from ._model import VolumeSize
+# XXX understand StoragePool and see if it can be made less ZFS specific.
+from .filesystems.openstack import StoragePool
+from ._model import VolumeSize, VolumeName
 from ..common.script import ICommandLineScript
+from flocker.volume.filesystems.openstack import driver_from_environment, get_public_ips, next_device
 
+# Add openstack API key here?
 DEFAULT_CONFIG_PATH = FilePath(b"/etc/flocker/volume.json")
 FLOCKER_MOUNTPOINT = FilePath(b"/flocker")
+# This is ZFS specific. Rename?
```

Comment:
No, it's not ZFS specific. Openstack volumes need a pool name too, which is used as the root beneath which volumes are mounted.